controllers/auth.js

```diff
@@ -0,0 +1,12 @@
+const User = require('../models/user')
+const { StatusCodes } = require('http-status-codes')
+
+const register = async (req, res) => {
+    const user = await User.create({ ...req.body })
```

Comment:
Could you just assign the fields that are required. 
i.e. 

    const user = await User.create({name: req.,body.name}) etc
    Its an easy way to get bugs by spreading as you may get something that wasnt meant to be in the request, and its also easier to see when reading the code, oh a user is constructed of this field etc